src/limiter/bound_preserving_limiter_factory.cpp

```diff
@@ -44,11 +53,16 @@ std::unique_ptr< BoundPreservingLimiter<dim, real> >
         return std::make_unique< MaximumPrincipleLimiter<dim, nstate, real> >(parameters_input);
     } else if (limiter_type == limiter_enum::positivity_preservingZhang2010
                 || limiter_type == limiter_enum::positivity_preservingWang2012) {
-        if (nstate == dim + 2)
+        if (nstate == dim + 2 && flux_nodes_type != flux_nodes_enum::GL)
             return std::make_unique< PositivityPreservingLimiter<dim, nstate, real> >(parameters_input);
         else {
-            std::cout << "Error: Cannot create Positivity-Preserving limiter for nstate_input != dim + 2" << std::endl;
-            std::abort();
+            if(nstate != dim + 2) {
+                std::cout << "Error: Cannot create Positivity-Preserving limiter for nstate_input != dim + 2" << std::endl;
+                std::abort();
+            } else if (flux_nodes_type == flux_nodes_enum::GL) {
```

Comment:
```suggestion
            } else if (flux_nodes_type != flux_nodes_enum::GLL) {
```

Reply:
is this more representative of the limitation? that it only works for GLL ? and not anything but GL

Reply:
It should work for any quadrature rule as long it includes the two cell ends, which GL does not. Since GL is the only one we have right now that doesn't include the two cell ends, I excluded only that.

Should I still commit the suggested change or should I leave a comment indicating that quadrature rules that don't include the two cell ends must be excluded?

Reply:
I think its safer to say it only works for GLL instead of anything but GL